fix for #220

lablnet opened the pull request:
https://github.com/zestframework/Zest_Framework/issues/220

lablnet committed:
fix for #220

lablnet committed:
refactor

peter279k reviewed (changes requested):
These classes seems to be done and my recommendations are as follows:

- Each archive classes should have their unit tests and it can make their behaviors verified and reliable.
- The StyleCI has been failed to check on the latest commit and it should be fixed.

lablnet committed:
refector

lablnet committed:
refector

Maikuolan committed:
Add an adapter interface. (#321)

When the intention exists to write numerous classes with the same or very
similar methods, PHP interfaces are a useful means to ensure that accidents
don't occur with method names in the future, and to ensure that such
intentions are fully met insofar as such classes possessing the needed
methods.

Maikuolan committed:
Fixes, improvements, tests (archive classes). (#322)

* Fix implements, add guard clauses, partially refactor.

- Fix interface implements.
- Add some guard clauses to the adapters.
- Partially refactors all affected code.
- Moves buffer declaration to a private property so that we don't need to
  repeat the variable multiple times in the code.

* Fix path issue.

Fixes an issue with bad paths being written to/from archives.

* An attempt to write some tests for the archive classes.

* Fix Sider and StyleCI complaints.

* Fix StyleCI complaint.

* Changes requested at PR.

Maikuolan reviewed (approved):
LGTM. :-)

peter279k commented on src/Archive/Archive.php:
```diff
@@ -0,0 +1,98 @@
+     *
+     * @return object
+     */
+    public function setAdapter($adapter)
```
To be consistency, it should add the native type hint:

```PHP
public function setAdapter($adapter): self
```

lablnet committed:
Update Archive.php
